### PR TITLE
feat: git merge branch action

### PR DESCRIPTION
**Pull request by thibthib18**
Hi! I noticed having to leave vim to merge branches, so I added the merge branch action to the git actions :)

Also added a default mapping `C-m` in `git.branches`.

### CODE REVIEW

File: doc/telescope.txt [hunk 1]
```diff
@@ -583,6 +583,7 @@ builtin.git_branches({opts})                          *builtin.git_branches()*
       - `<C-r>`: rebases currently selected branch
       - `<C-a>`: creates a new branch, with confirmation prompt before creation
       - `<C-d>`: deletes the currently selected branch, with confirmation
+      - `<C-m>`: merges the currently selected branch, with confirmation
```
Comment (Conni2461) [inline doc/telescope.txt:586]: You need to change this here: https://github.com/nvim-telescope/telescope.nvim/blob/master/lua/telescope/builtin/init.lua#L171-L179

doc/telescope.txt is 100% autogenerated

File: lua/telescope/actions/init.lua [hunk 2]
```diff
@@ -517,6 +517,28 @@ actions.git_delete_branch = function(prompt_bufnr)
   end
 end
 
+--- Merge the currently selected branch
+---@param prompt_bufnr number: The prompt bufnr
+actions.git_merge_branch = function(prompt_bufnr)
```
Comment (Conni2461) [inline lua/telescope/actions/init.lua:522]: the branch actions are all very similar. Maybe a local function that takes args_tbl, question, success, failed message. To reduce code duplication

File: doc/telescope.txt (hunk 1 above)
Comment (thibthib18) [reply doc/telescope.txt]: Thanks I wasn't sure which one to update :)

Updated

File: lua/telescope/actions/init.lua (hunk 2 above)
Comment (thibthib18) [reply lua/telescope/actions/init.lua]: Agree, updated.

**Comment by thibthib18** [comment]
It seems that the pipeline is failing partly due to styling (mostly `'` replaced with `"`), how can I run the style checker locally?

**Comment by fdschmidt93** [comment]
Probably easiest just to format with our stylua [config](https://github.com/nvim-telescope/telescope.nvim/blob/master/.stylua.toml)

File: lua/telescope/builtin/git.lua [hunk 3]
```diff
@@ -297,6 +297,9 @@ git.branches = function(opts)
 
       map("i", "<c-d>", actions.git_delete_branch)
       map("n", "<c-d>", actions.git_delete_branch)
+
+      map("i", "<c-m>", actions.git_merge_branch)
+      map("n", "<c-m>", actions.git_merge_branch)
```
Comment (Conni2461) [inline lua/telescope/builtin/git.lua:302]: `<c-m>` is not a good mapping. See `:help <cr>`. `<c-m>` and `<cr>` are the same meaning that we no longer can checkout a different branch with this PR

**Comment by Conni2461** [comment]
Otherwise LGTM. I generate documentation because docgen CI doesnt always work. Just change the mapping to something else in both code and doc. Sadly i dont have a different suggestion.

Thanks for the PR :)

File: lua/telescope/builtin/git.lua (hunk 3 above)
Comment (thibthib18) [reply lua/telescope/builtin/git.lua:302]: Ooh this is why I had so much trouble using it at first! Damn I lost at least 15 minutes on this. 
Will update it rn

**Comment by thibthib18** [comment]
Not easy to pick one indeed haha. Went for `Y` cause it looks like two branch being merged :stuck_out_tongue:

File: lua/telescope/builtin/init.lua [hunk 4]
```diff
@@ -175,6 +175,7 @@ builtin.git_bcommits = require("telescope.builtin.git").bcommits
 ---   - `<C-r>`: rebases currently selected branch
 ---   - `<C-a>`: creates a new branch, with confirmation prompt before creation
 ---   - `<C-d>`: deletes the currently selected branch, with confirmation prompt before deletion
+---   - `<C-m>`: merges the currently selected branch, with confirmation prompt before deletion
```
Comment (fdschmidt93) [inline lua/telescope/builtin/init.lua:178]: You should only change the mapping for docs here from `<C-m>` to `<C-y>`. telescope.txt is auto-generated from the annotations with CI.

Not sure, you might have git conflicts now at your hand if docgen is triggered :sweat_smile:

Comment (Conni2461) [reply lua/telescope/builtin/init.lua:178]: he should just change it everywhere. CI isnt running thats why i manually ran it. Sometimes that happens for new contributors.

He should also change it in init.lua so it doesnt break in future PRs.

Comment (thibthib18) [reply lua/telescope/builtin/init.lua]: I updated it, lets see if conflicts arise or not :crossed_fingers:

**Review by Conni2461 — APPROVED** [review]
Thanks :)

**Pull request merged**